src/driver/driver.cpp

```diff
@@ -205,25 +205,60 @@ void EvolutionDriver::InitializeBlockTimeSteps() {
 // \brief function that loops over all MeshBlocks and find new timestep
 
 void EvolutionDriver::SetGlobalTimeStep() {
-  // don't allow dt to grow by more than 2x
-  // consider making this configurable in the input
-  if (tm.dt < 0.1 * std::numeric_limits<Real>::max()) {
-    tm.dt *= 2.0;
-  }
-  Real big = std::numeric_limits<Real>::max();
-  for (auto const &pmb : pmesh->block_list) {
-    tm.dt = std::min(tm.dt, pmb->NewDt());
-    pmb->SetAllowedDt(big);
+  // Check if user wants to force the value
+  if (dt_force > 0.0) {
+    tm.dt = dt_force;
+  } else {
+    // Check for special first cycle value
+    if (tm.ncycle == 0) {
+      tm.dt = std::min(tm.dt, dt_init);
+    }
+    // don't allow dt to grow by more than 2x
+    // consider making this configurable in the input
+    if (tm.dt < 0.1 * std::numeric_limits<Real>::max()) {
+      tm.dt *= 2.0;
+    }
+    // Allow the meshblocks to vote
+    Real big = std::numeric_limits<Real>::max();
+    for (auto const &pmb : pmesh->block_list) {
+      tm.dt = std::min(tm.dt, pmb->NewDt());
+      pmb->SetAllowedDt(big);
+    }
+    // Allow the user to vote
+    tm.dt = std::min(tm.dt, dt_user);
   }
 
 #ifdef MPI_PARALLEL
   PARTHENON_MPI_CHECK(MPI_Allreduce(MPI_IN_PLACE, &tm.dt, 1, MPI_PARTHENON_REAL, MPI_MIN,
                                     MPI_COMM_WORLD));
 #endif
 
-  if (tm.time < tm.tlim &&
-      (tm.tlim - tm.time) < tm.dt) // timestep would take us past desired endpoint
+  // Check that we have not gone off the rails
+  if (tm.dt <= dt_min) {
+    if (++dt_min_count >= dt_min_count_max) {
+      std::stringstream msg;
+      msg << "Timesetep has fallen bellow minimum (parthenon/time/dt_min=" << dt_min
+          << ") for more than " << dt_min_count_max << " steps";
+      PARTHENON_FAIL(msg);
+    }
+  } else {
+    dt_min_count = 0;
+  }
+  if (tm.dt >= dt_max) {
+    if (++dt_max_count >= dt_max_count_max) {
+      std::stringstream msg;
+      msg << "Timesetep has risen above maximum (parthenon/time/dt_max=" << dt_max
+          << ") for more than " << dt_max_count_max << " steps";
+      PARTHENON_FAIL(msg);
+    }
```

Comment:
Why not just limit the step to `dt_max`?

Reply:
I debated whether to add new parameters called `dt_floor` and `dt_ceil` such that `dt = st::max(dt_floor, std::min(dt, dt_ceil))`, which does what you're suggesting. Those are separate from `dt_min` and `dt_max`. 

Sims may just blow up in timestep, and `dt_max` is meant to catch that and die. It's not really mean to limit your dead simulation to a large timestep value -- but `dt_ceil` could do that.

Reply:
Maybe it makes sense to rename as `dt_max_allowed` and `dt_min_allowed` so that it is clear that they aren't a floor and ceiling? I agree that there probably isn't much use for the floor/ceil since hitting those generally implies something isn't going so well in the sim. Just thought maybe we should add them for completeness.

Reply:
Went ahead and added `dt_floor` and `dt_ceil`